From 69a132111ecc8869477934cf22e06e79a25ddce3 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 24 Jul 2026 21:39:18 +0800
Subject: [PATCH 441/450] AOSCOS: mmc: loongson2: add back the so-called "ABI
 1.0" compatible naming for 2K2000/2K3000/3B6000M

Older Loongson 2K3000/3B6000M firmware uses the following so-called "ABI
1.0" compatible name. Add this back for compatibility's sake.

Update your firmware and care for your customer's right! Ugh.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 drivers/mmc/host/loongson2-mmc.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/mmc/host/loongson2-mmc.c b/drivers/mmc/host/loongson2-mmc.c
index 118eccbaf8cde..7db53ac231f98 100644
--- a/drivers/mmc/host/loongson2-mmc.c
+++ b/drivers/mmc/host/loongson2-mmc.c
@@ -1014,6 +1014,12 @@ static const struct of_device_id loongson2_mmc_of_ids[] = {
 	{ .compatible = "loongson,ls2k0500-mmc", .data = &ls2k0500_mmc_pdata },
 	{ .compatible = "loongson,ls2k1000-mmc", .data = &ls2k1000_mmc_pdata },
 	{ .compatible = "loongson,ls2k2000-mmc", .data = &ls2k2000_mmc_pdata },
+	/*
+	 * Older Loongson 2K3000/3B6000M firmware uses the following so-called
+	 * "ABI 1.0" compatible name. Add this back for compatibility's sake...
+	 * Update your firmware and care for your customer's right! Ugh.
+	 */
+	{ .compatible = "loongson,ls2k_sdio_1.2", .data = &ls2k2000_mmc_pdata },
 	{ },
 };
 MODULE_DEVICE_TABLE(of, loongson2_mmc_of_ids);
-- 
2.55.0

